pallene/coder.lua

```diff
@@ -790,6 +794,22 @@ function RecordCoder:declare_struct()
     return out
 end
 
+function RecordCoder:declare_getmem()
+    if self.layout.prim_size == 0 then return "" end
+
+    local out = util.render([[
+        static ${STRUCT} *${GETMEM}(Udata *u)
+        {
+            return (${STRUCT} *)(cast_charp(u) + udatamemoffset(${NUVALUE}));
+        }
+    ]], {
+        STRUCT = self:struct(),
+        GETMEM = self:getmem(),
+        NUVALUE = self.layout.gc_size,
+    })
+    return out
+end
+
```

Comment:
I think it would be helpful to add a small comment pointing out that this is a specialization of getudatamem that generates cleaner assembly code because NUVALUE is a compile-time constant.